test/extended/clusterup.sh

```diff
@@ -0,0 +1,245 @@
+    export HOME="${test_home}"
+    pushd "${HOME}" &> /dev/null
+    os::log::info "Using ${HOME} as home directory"
+    ${funcname}
+    popd &> /dev/null
+    export HOME="${global_home}"
+
+    os::log::info "cluster up -- ${test}: ok"
+    os::test::junit::declare_suite_end
+}
+
+function os::test::extended::clusterup::verify_router_and_registry () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get svc -n default" "docker-registry"
+    os::cmd::expect_success_and_text "oc get svc -n default" "router"
+	os::cmd::try_until_text "oc get endpoints docker-registry -o jsonpath='{ .subsets[*].ports[?(@.name==\"5000-tcp\")].port }' -n default" "5000" $(( 10*minute )) 1
+	os::cmd::try_until_text "oc get endpoints router -o jsonpath='{ .subsets[*].ports[?(@.name==\"80-tcp\")].port }' -n default" "80" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_image_streams () {
+    os::cmd::try_until_text "oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'" "latest" $(( 10*minute )) 1
+}
+
+function os::test::extended::clusterup::verify_ruby_build () {
+    os::cmd::expect_success "oc new-app https://github.com/openshift/ruby-hello-world.git"
+    os::cmd::try_until_text "oc get builds -l app=ruby-hello-world -o jsonpath='{ .items[*].status.phase }'" "Complete" $(( 10*minute )) 1
+}
+
+function os::test::extended::clusterup::verify_persistent_volumes () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get jobs -n default" "persistent-volume-setup"
+    os::cmd::try_until_text "oc get job persistent-volume-setup -n default -o jsonpath='{ .status.succeeded }'" "1" $(( 10*minute )) 1
+    os::cmd::expect_success "oc get pv/pv0100"
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_metrics () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get pods -n openshift-infra" "metrics-deployer"
+    os::cmd::try_until_text "oc get pods -n openshift-infra -l job-name=metrics-deployer-pod" "Completed" $(( 20*minute )) 2
+	os::cmd::try_until_text "oc get endpoints hawkular-metrics -o jsonpath='{ .subsets[*].ports[?(@.name==\"443-tcp\")].port }' -n openshift-infra" "443" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_logging () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get pods -n logging" "logging-deployer"
+    os::cmd::try_until_text "oc get pods -n logging -l logging-infra=deployer" "Completed" $(( 20*minute )) 2
+	os::cmd::try_until_text "oc get endpoints logging-kibana -o jsonpath='{ .subsets[*].ports[?(@.name==\"3000-tcp\")].port }' -n logging" "3000" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
```

Comment:
So I tried running on my Mac with `--service-catalog` and I get this:
```
oc cluster up --service-catalog
Starting OpenShift using openshift/origin:v3.6.0-alpha.2 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v3.6.0-alpha.2 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ... 
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 127.0.0.1 as the server IP
-- Starting OpenShift container ... 
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Adding default OAuthClient redirect URIs ... OK
-- Installing registry ... OK
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Installing service catalog ... FAIL
   Error: cannot instantiate service catalog template
   Caused By:
     Error: cannot create objects from template openshift/service-catalog
     Caused By:
       Error: [policybindings.authorization.openshift.io "service-catalog:default" not found, policybindings.authorization.openshift.io "kube-system:default" not found]
```
The outcome is the same whether using this branch or master. I'll create an issue for it and when it's fixed, it should be very easy to add to the extended test.